[zipkin-lens] Add tooltip for service names

rahuliyer95 opened the pull request:
Added a simple tooltip for service names.

![Screenshot 2020-07-03 at 11 22 28 AM](https://user-images.githubusercontent.com/4255590/86491592-f0371900-bd1f-11ea-95a0-53da28bd6c3c.png)

rahuliyer95 committed:
[zipkin-lens] Add tooltip for service names

codefromthecrypt commented:
Hi! thanks for helping. I wonder what is your motivation.
 is it a very long service name? if so can you show a screenshot of where this helps?

rahuliyer95 commented:
Yes, should be useful for viewing the full name for services that have long names. An example screenshot 
<img width="788" alt="Screenshot 2020-07-03 at 9 36 49 PM" src="https://user-images.githubusercontent.com/4255590/86505089-8184aa80-bd75-11ea-9984-9bd4e3e5b826.png">

codefromthecrypt commented:
makes sense. @openzipkin/ui anyone have other ideas on how to handle long service names? I admit I've also wondered about this.

codefromthecrypt commented on zipkin-lens/src/components/TracePage/TraceTimeline/TraceTree.jsx:
```diff
@@ -294,6 +294,7 @@ const TraceTree = React.memo(
                 className={classes.serviceBadgeText}
               >
                 {serviceName}
+                <title>{serviceName}</title>
```
deserves a comment, unit test or both

adriancole committed:
Removes guava dep in preparation of Datastax Driver 4.x (#3134)

We only needed guava for Cassandra. Removing it here makes migration easier.

Note: there's still a guava left in cassandra-v1 which is too complex to undo
in this PR (Table and Cache apis).

adriancole committed:
Removes guava Table dep from cassandra-v1 (#3135)

adriancole committed:
Refactors DelayLimiter in preparation of re-use in cassandra v1 (#3136)

This refactors code relating to DelayLimiter so that it is easier to
reuse later. What will happen in a separate PR (or what will be
attempted) is change the limiter so that it can run a predicate instead
of an exact match. For example Cassandra v1's Indexer currently mutes
index insertions which are logically equivalent to data already written.

Concretely, a service name that appears 3 times in the same trace,
results in two writes (earliest and latest). If it appears again, and
between those timestamps, that write is muted as it is redundant.

tacigar commented:
I think tooltip is a good idea.

tacigar commented:
But I cannot see tooltip in my chrome(83.0.4103.116)... 🤔

jorgheymans commented:
Consider also #3118 and #2948

rahuliyer95 commented:
I added a small comment about the tooltip and referenced it to the doc for the title element. I am not an expert here so I would love some guidance around the unit-test. I believe #3118 and #2948 are more involved and I would prefer tackling them separately and not as a part of this PR.

Commit:
[zipkin-lens] Lint fixes

Commit:
[zipkin-lens] Add a small comment explaning the tooltip

Commit:
Merge remote-tracking branch 'upstream/master' into feature/tooltip-for-servicename

codefromthecrypt commented on zipkin-lens/src/components/TracePage/TraceTimeline/TraceTree.jsx:
```diff
@@ -294,6 +294,13 @@ const TraceTree = React.memo(
                 className={classes.serviceBadgeText}
               >
                 {serviceName}
+                <title>
```
so this is all wrapped in `React.memo(` I suspect it is harder to test, does anyone know if we have an example that tests something like this?

we can merge, just so far one person said they cannot see tool tip as a result at the moment.. cc @openzipkin/ui

jorgheymans replied:
Tooltips fine here on latest FF Ubuntu. 

@rahuliyer95 Wondering if this needs to be tested on a larger trace with a couple of thousand spans, would the implementation suffer from this? I don't have have such trace handy here but maybe it would be good to test this.

codefromthecrypt replied:
@drolando mind testing this for impact ^^ I think we don't have a massive test trace

codefromthecrypt commented:
@hanahmily @beckje01 do you have large trace with large service names? We are looking for some feedback to make sure that the tooltip works, but also doesn't slow rendering etc.
